### PR TITLE
utils/image.py: fetch_image() scrape from Tenor

### PR DESCRIPTION
**Pull request by h-anjru**
Adds a conditional to `fetch_image()` in `utils/image.py` which looks for a Tenor link--which display in Discord as a GIF despite being a link to a Web page--and, if a Tenor link, then scrapes the GIF URL from the Tenor page. A new dependency, BeautifulSoup, has been added to help pull data out of the fetched HTML.

### CODE REVIEW

File: crimsobot/utils/image.py [hunk 1]
```diff
@@ -115,6 +116,12 @@ async def fetch_image(ctx: Context, arg: Optional[str]) -> Image.Image:
     session = aiohttp.ClientSession()
 
     async def open_img_from_url(url: str) -> Image.Image:
+        if 'tenor.com/view' in url:
+            async with session.get(url, allow_redirects=False) as response:
```
Comment (h-anjru) [inline crimsobot/utils/image.py:120]: This is WET. I shouldn't need two `session.get()`s to get the job done here, but I can't quite figure out how to do this with a single `session.get()`. I blame my unfamiliarity with aiohttp.

Comment (williammck) [reply crimsobot/utils/image.py:120]: nah it's fine. you have to make two requests anyway, no way around it

File: crimsobot/utils/image.py [hunk 2]
```diff
@@ -115,6 +116,12 @@ async def fetch_image(ctx: Context, arg: Optional[str]) -> Image.Image:
     session = aiohttp.ClientSession()
 
     async def open_img_from_url(url: str) -> Image.Image:
+        if 'tenor.com/view' in url:
```
Comment (williammck) [inline crimsobot/utils/image.py:119]: someone could possibly put tenor.com/view in a random other url but idk what they'd really achieve by that, just wanted to at least call it out

**Review by williammck — APPROVED** [review]

File: crimsobot/utils/image.py (hunk 2 above)
Comment (h-anjru) [reply crimsobot/utils/image.py:119]: "Congratulations, you made a Discord bot return an error! You are very intelligent."

**Pull request merged**